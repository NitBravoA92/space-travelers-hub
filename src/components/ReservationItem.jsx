import PropTypes from 'prop-types';

const ReservationItem = ({ serviceName }) => (
  <li className="reservation-item">
    {serviceName}
  </li>
);

ReservationItem.propTypes = {
  serviceName: PropTypes.string.isRequired,
};

export default ReservationItem;
