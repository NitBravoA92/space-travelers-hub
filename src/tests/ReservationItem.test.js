import '@testing-library/jest-dom/extend-expect';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import ReservationItem from '../components/ReservationItem';

describe('The ReservationItem component', () => {
  test('renders into the DOM', () => {
    const component = renderer
      .create(
        <ReservationItem serviceName="Falcon 1" />,
      ).toJSON();
    expect(component).toMatchSnapshot();
  });

  test("render a li element with the text 'Thaicom'", () => {
    const component = render(<ReservationItem serviceName="Thaicom" />);

    const liElement = component.container.querySelector('li');

    expect(liElement).toBeInTheDocument();
  });
});
