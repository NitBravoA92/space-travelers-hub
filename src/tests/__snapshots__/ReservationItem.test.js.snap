// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The ReservationItem component renders into the DOM 1`] = `
<li
  className="reservation-item"
>
  Falcon 1
</li>
`;
